doom/config.org
#+title: Config
#+PROPERTY: header-args:emacs-lisp :tangle config.el

* Basic configurations
** Default directory
#+begin_src emacs-lisp
(setq org-directory "~/org/")
#+end_src

** Line number
#+begin_src emacs-lisp
(setq display-line-numbers-type 'relative)
#+end_src

** Themes
#+begin_src emacs-lisp
(setq doom-theme 'doom-one)
;; (setq doom-theme 'doom-one-light)
#+end_src

** Font
#+begin_src emacs-lisp
;;(setq doom-font (font-spec :family "Fira Code" :size 13))
(setq doom-font (font-spec :family "Hack" :size 13))
#+end_src

* Org
** Keybindings
#+begin_src emacs-lisp
(after! doom
  (map! :leader
        "o" nil))
(after! doom
  (map! :leader
        (:prefix ("o" . "Org")
         (:prefix ("m" . "Mode")
          ;; Folding
          "TAB"     #'org-cycle
          "a"       #'org-shifttab

          ;; Headings
          "h"       #'org-meta-return
          "H"       #'org-insert-todo-heading

          ;; TODOs
          "t"       #'org-todo

          ;; Structure
          "<left>"  #'org-metaleft
          "<right>" #'org-metaright
          "<up>"    #'org-metaup
          "<down>"  #'org-metadown

          ;; Dates
          "s"       #'org-schedule
          "d"       #'org-deadline
          "."       #'org-time-stamp

          ;; Links
          "l"       #'org-insert-link
          "o"       #'org-open-at-point

          ;; Tags
          "q"       #'org-set-tags-command

          ;; Archive
          "x"       #'org-archive-subtree

          ;; Context-sensitive
          "c"       #'org-ctrl-c-ctrl-c

          ;; Export
          "e"       #'org-export-dispatch

          ;; Source blocks
          "'"       #'org-edit-special

          ;; Narrowing
          "n"       #'org-narrow-to-subtree
          "w"       #'widen))))
#+end_src

** Org capture
*** Keybindings
#+begin_src emacs-lisp
(map! :leader
      :desc "org-capture" "c" #'org-capture)
#+end_src

*** Templates
#+begin_src emacs-lisp
(after! org
  (setq org-capture-templates
        '(
          ("t" "tasks" entry
           (file "~/org/tasks.org")
           "* TODO %? :todos:\n:PROPERTIES:\n:created: %U\n:END:\n")
          ("n" "notes" entry
           (file "~/org/notes.org")
           "* NOTE %? :notes:\n:PROPERTIES:\n:created: %U\n:END:\n")
          ("i" "ideas" entry
           (file "~/org/ideas.org")
           "* IDEA %? :ideas:\n:PROPERTIES:\n:created: %U\n:END:\n"))))

;;          ("j" "journal")
;;          ("jr" "reflections" entry
;;           (file+olp+datetree "~/org/journal.org")
;;           "* %U Reflection %? :reflections:\n")
;;          ("jo" "observations" entry
;;           (file+olp+datetree "~/org/journal.org")
;;           "* %U %? :observations:\n")
#+end_src

** Org todo keywords
#+begin_src emacs-lisp
(after! org
  (setq org-todo-keywords
        '((sequence "PROJ(p)" "PLAN(l)" "PROG(g)" "WAIT(w)" "HOLD(h)" "|" "COMP(m)" "CANC(c)")
          (sequence "TODO(t)" "STRT(s)" "WAIT(w)" "HOLD(h)" "|" "DONE(d)" "KILL(k)")
          (sequence "NOTE(n)" "STRT(s)" "WAIT(w)" "HOLD(h)" "|" "ARCH(a)" "KILL(k)")
          (sequence "IDEA(i)" "STRT(s)" "WAIT(w)" "HOLD(h)" "|" "PROC(r)" "KILL(k)"))))
#+end_src

** Org agenda
*** Configuration
#+begin_src emacs-lisp
(setq org-agenda-files
      (directory-files-recursively "~/org/" "\\.org$"))

(setq org-agenda-skip-scheduled-if-done t
      org-agenda-skip-deadline-if-done t)
#+end_src

*** Keybindings
#+begin_src emacs-lisp
(map! :leader
      :desc "org-agenda" "a" #'org-agenda)
#+end_src
*** Sort
#+begin_src emacs-lisp
(setq org-agenda-sorting-strategy
      '((tags tag-up)))
#+end_src

** Org roam
*** Configuration
#+begin_src emacs-lisp
(use-package! org-roam
              :custom
              (org-roam-directory (file-truename "~/org/"))
              :config
              (org-roam-db-autosync-mode)
              (map! :leader
                   (:prefix ("r" . nil)
                    :desc "Org roam capture"                "c"   #'org-roam-capture
                    :desc "Org roam insert node"            "i"  #'org-roam-node-insert
                    :desc "Org roam find node"              "f"  #'org-roam-node-find
                    :desc "Org roam UI"                     "g"  #'org-roam-ui-mode
                    :desc "Org roan dailies goto today"     "t" #'org-roam-dailies-capture-today
                    :desc "Org roam dailies goto tomorrow"  "T" #'org-roam-dailies-goto-tomorrow
                    :desc "Org roam dailies goto yesterday" "y" #'org-roam-dailies-goto-yesterday)))
(use-package org-roam-dailies
             :after org-roam
             :config
             (setq org-roam-dailies-directory "roam/dailies/"))
#+end_src

*** Templates
#+begin_src emacs-lisp
(setq org-roam-capture-templates
      '(("n" "note" plain "%?"
         :target (file+head "roam/${slug}.org"
                            "#+title: ${title}\n#+created: %U\n#+filetags: \n")
         :unnarrowed t)))

(setq org-roam-dailies-capture-templates
      '(("d" "dailies" entry "* %?"
         :target (file+head "%<%Y-%m-%d>.org"
                            "#+title: %<%Y-%m-%d>\n"))))
#+end_src

*** Org roam ui
#+begin_src emacs-lisp
(use-package! org-roam-ui
              :after org-roam
              :hook (org-roam-mode . org-roam-ui-mode)
              :config
              (setq org-roam-ui-sync-theme t
                   org-roam-ui-follow t
                    org-roam-ui-update-on-save t
                    org-roam-ui-open-on-start t))
#+end_src

* Misc.
** Vim bindings
#+begin_src emacs-lisp
(setq evil-escape-key-sequence "jk")
(setq evil-escape-delay 0.2)
#+end_src

** Hunspell
#+begin_src emacs-lisp
(setq ispell-program-name "hunspell")

(setq ispell-hunspell-dict-paths-alist
      '(("en_US" "C:/Hunspell/dict/en_US.aff")))

(setq ispell-local-dictionary "en_US")
#+end_src

** Bin
#+begin_src emacs-lisp
(setq delete-by-moving-to-trash t)
#+end_src

** Confirm exist
#+begin_src emacs-lisp
(setq confirm-kill-emacs nil)
#+end_src

** Initial buffer
#+begin_src emacs-lisp
(setq initial-buffer-choice "~/org/")
#+end_src
** Unfold drawer
#+begin_src emacs-lisp
(after! org
  (setq org-cycle-hide-drawer-startup nil))
#+end_src


doom/config.el
(setq org-directory "~/org/")

(setq display-line-numbers-type 'relative)

(setq doom-theme 'doom-one)
;; (setq doom-theme 'doom-one-light)

;;(setq doom-font (font-spec :family "Fira Code" :size 13))
(setq doom-font (font-spec :family "Hack" :size 13))

(after! doom
  (map! :leader
        "o" nil))
(after! doom
  (map! :leader
        (:prefix ("o" . "Org")
         (:prefix ("m" . "Mode")
          ;; Folding
          "TAB"     #'org-cycle
          "a"       #'org-shifttab

          ;; Headings
          "h"       #'org-meta-return
          "H"       #'org-insert-todo-heading

          ;; TODOs
          "t"       #'org-todo

          ;; Structure
          "<left>"  #'org-metaleft
          "<right>" #'org-metaright
          "<up>"    #'org-metaup
          "<down>"  #'org-metadown

          ;; Dates
          "s"       #'org-schedule
          "d"       #'org-deadline
          "."       #'org-time-stamp

          ;; Links
          "l"       #'org-insert-link
          "o"       #'org-open-at-point

          ;; Tags
          "q"       #'org-set-tags-command

          ;; Archive
          "x"       #'org-archive-subtree

          ;; Context-sensitive
          "c"       #'org-ctrl-c-ctrl-c

          ;; Export
          "e"       #'org-export-dispatch

          ;; Source blocks
          "'"       #'org-edit-special

          ;; Narrowing
          "n"       #'org-narrow-to-subtree
          "w"       #'widen))))

(map! :leader
      :desc "org-capture" "c" #'org-capture)

(after! org
  (setq org-capture-templates
        '(
          ("t" "tasks" entry
           (file "~/org/tasks.org")
           "* TODO %? :todos:\n:PROPERTIES:\n:created: %U\n:END:\n")
          ("n" "notes" entry
           (file "~/org/notes.org")
           "* NOTE %? :notes:\n:PROPERTIES:\n:created: %U\n:END:\n")
          ("i" "ideas" entry
           (file "~/org/ideas.org")
           "* IDEA %? :ideas:\n:PROPERTIES:\n:created: %U\n:END:\n"))))

;;          ("j" "journal")
;;          ("jr" "reflections" entry
;;           (file+olp+datetree "~/org/journal.org")
;;           "* %U Reflection %? :reflections:\n")
;;          ("jo" "observations" entry
;;           (file+olp+datetree "~/org/journal.org")
;;           "* %U %? :observations:\n")

(after! org
  (setq org-todo-keywords
        '((sequence "PROJ(p)" "PLAN(l)" "PROG(g)" "WAIT(w)" "HOLD(h)" "|" "COMP(m)" "CANC(c)")
          (sequence "TODO(t)" "STRT(s)" "WAIT(w)" "HOLD(h)" "|" "DONE(d)" "KILL(k)")
          (sequence "NOTE(n)" "STRT(s)" "WAIT(w)" "HOLD(h)" "|" "ARCH(a)" "KILL(k)")
          (sequence "IDEA(i)" "STRT(s)" "WAIT(w)" "HOLD(h)" "|" "PROC(r)" "KILL(k)"))))

(setq org-agenda-files
      (directory-files-recursively "~/org/" "\\.org$"))

(setq org-agenda-skip-scheduled-if-done t
      org-agenda-skip-deadline-if-done t)

(map! :leader
      :desc "org-agenda" "a" #'org-agenda)

(setq org-agenda-sorting-strategy
      '((tags tag-up)))

(use-package! org-roam
              :custom
              (org-roam-directory (file-truename "~/org/"))
              :config
              (org-roam-db-autosync-mode)
              (map! :leader
                   (:prefix ("r" . nil)
                    :desc "Org roam capture"                "c"   #'org-roam-capture
                    :desc "Org roam insert node"            "i"  #'org-roam-node-insert
                    :desc "Org roam find node"              "f"  #'org-roam-node-find
                    :desc "Org roam UI"                     "g"  #'org-roam-ui-mode
                    :desc "Org roan dailies goto today"     "t" #'org-roam-dailies-capture-today
                    :desc "Org roam dailies goto tomorrow"  "T" #'org-roam-dailies-goto-tomorrow
                    :desc "Org roam dailies goto yesterday" "y" #'org-roam-dailies-goto-yesterday)))
(use-package org-roam-dailies
             :after org-roam
             :config
             (setq org-roam-dailies-directory "roam/dailies/"))

(setq org-roam-capture-templates
      '(("n" "note" plain "%?"
         :target (file+head "roam/${slug}.org"
                            "#+title: ${title}\n#+created: %U\n#+filetags: \n")
         :unnarrowed t)))

(setq org-roam-dailies-capture-templates
      '(("d" "dailies" entry "* %?"
         :target (file+head "%<%Y-%m-%d>.org"
                            "#+title: %<%Y-%m-%d>\n"))))

(use-package! org-roam-ui
              :after org-roam
              :hook (org-roam-mode . org-roam-ui-mode)
              :config
              (setq org-roam-ui-sync-theme t
                   org-roam-ui-follow t
                    org-roam-ui-update-on-save t
                    org-roam-ui-open-on-start t))

(setq evil-escape-key-sequence "jk")
(setq evil-escape-delay 0.2)

(setq ispell-program-name "hunspell")

(setq ispell-hunspell-dict-paths-alist
      '(("en_US" "C:/Hunspell/dict/en_US.aff")))

(setq ispell-local-dictionary "en_US")

(setq delete-by-moving-to-trash t)

(setq confirm-kill-emacs nil)

(setq initial-buffer-choice "~/org/")

(after! org
  (setq org-cycle-hide-drawer-startup nil))
